plugins/org.eclipse.n4js.ide/src/org/eclipse/n4js/ide/xtext/server/DebugService.java

```diff
@@ -69,10 +89,90 @@ public void connect(LanguageClient client) {
 		public CompletableFuture<Void> printDebugInfo() {
 			return CompletableFuture.failedFuture(new UnsupportedOperationException());
 		}
+
+		@Override
+		public String getDebugInfo() {
+			return null;
+		}
+	}
+
+	/** Abstract base implementation for debug services that intend to include LSP message tracing. */
+	@Singleton
+	public abstract class AbstractTracingDebugService implements DebugService {
```

Comment:
Consider to remove the dependency to DebugService since it is not necessary.

Reply:
Now using containment instead of inheritance.